### ISSUE TITLE
TcpStream::connect should require crate feature 'rt'.

### ISSUE THREAD

**Issue by HeikoRibberink** [tokio-rs/tokio#5147]
**Version**
```
tokio v1.21.2
├── bytes v1.2.1
├── memchr v2.5.0
├── mio v0.8.4
│   ├── log v0.4.17
│   │   └── cfg-if v1.0.0
│   └── windows-sys v0.36.1
│       └── windows_x86_64_msvc v0.36.1
├── num_cpus v1.13.1
├── pin-project-lite v0.2.9
├── socket2 v0.4.7
│   └── winapi v0.3.9
└── winapi v0.3.9
```

**Platform**
OS Name:                   Microsoft Windows 10 Home
OS Version:                10.0.19043 N/A Build 19043

**Description**
When calling `tokio::io::TcpStream::connect` from outside a the context of a Tokio Runtime (in my case, a Bevy Taskpool), it panics with `there is no reactor running, must be called from the context of a Tokio 1.x runtime`. As the Tokio Runtime requires the crate feature 'rt', this function should also require that feature.

This code sample should be able to reproduce the bug.
```rs
use bevy::prelude::*;
use bevy::tasks::{IoTaskPool, Task, TaskPool};
use tokio::io::AsyncWriteExt;
use tokio::net::TcpStream;
fn main() {
    App::new()
        .add_plugins(DefaultPlugins)
        .add_startup_system(startup)
        .run();
    fn startup() {
        let task_pool = IoTaskPool::init(|| TaskPool::new());
        task_pool
            .spawn(async {
                let mut stream = TcpStream::connect("172.0.0.0:8080").await.unwrap();
                loop {
                    stream.write_u8(0).await.unwrap();
                }
            })
            .detach();
    }
}
```
If you run this with `RUST_BACKTRACE=1`, you will see a call to `tokio::runtime::io::Handle::current`, which panics as it is not inside a Runtime context.

I do not know what the advantage is of fixing this, as the panic message is pretty clear.

Thank you!

**Comment by Darksonn** [comment]
None of the types in `tokio::net` work without the `rt` feature. We could debate whether they should exist when the `rt` feature is disabled, but we cannot change it due to backwards compatibility, so the point is moot.

Closing as wontfix.
